SETUP/GRAPH.md

```diff
@@ -0,0 +1,138 @@
+# Graphs
+
+This file contains a description of the way graphs are created in dproofreaders.
+
+In case of problems, please post to the [DP Site Code](http://www.pgdp.net/phpBB3/viewforum.php?f=32)
+forum at the [primary DP site](https://www.pgdp.net/c).
+
+## Overview
+Graphs are drawn on the browser using a set of functions from [svgGraphs.js](../scripts/svgGraphs.js) that interact with the [d3js](https://d3js.org/) library to render [Scalable Vector Graphics (svg)](https://en.wikipedia.org/wiki/Scalable_Vector_Graphics) elements. The 4 types of graphs that are supported are
+1. Bar
+1. Line
+1. Pie
+1. Stacked Area
+
+These graphs are configured using a plain old javascript object. To facilitate creating the javascript to render a graph from PHP, a php utility method [build_svg_graph_inits](../pinc/graph_data.inc) was created. It is a function that takes an array of arrays representing the graphs to render. Each graph to render is a PHP array where the entries are
+1. type - type of graph to be rendered. Either `barLineGraph`, `pieGraph`, or `stackedAreaGraph`
+1. id - id of the div on the page to render the graph into
+1. configuration - PHP array configuring the graph. Expanded on below.
+Example Usage:
+
+```
+build_svg_graph_inits([["barLineGraph", "myGraphId", $graphConfig]]);
+```
+
+This will give you a string you can use to pass to js_data in the [output_header](../pinc/theme.inc) function. Something like
+
+```
+"$(function(){barLineGraph('myGraphId', graphConfigEncodedAsJSON)});"
+```
+
+# Configuration
+Each graph type supports a configuration object that allows control of the appearance of the graph as well as the data to the graph. Each graph type respects different configuration properties.
+## Bar/Line Graph
+```
```

Comment:
Consider ` ```js ` (at least it looks like valid JS to me?)

Reply:
It is essentially typescript so I can define the shapes of the objects. I'll annotate it as such.